service/src/main/java/bio/terra/workspace/service/iam/SamService.java

```diff
@@ -420,6 +421,28 @@ public void deleteWorkspace(AuthenticatedUserRequest userRequest, UUID uuid)
                 + "This just means the deletion was tried twice so no error thrown.");
         return;
       }
+
+      // Diagnostic: if the error was "Cannot delete a resource with children" then fetch and
+      // dump the remaining children.
+      if (apiException.getCode() == HttpStatus.BAD_REQUEST.value()
+          && StringUtils.contains(
+              apiException.getMessage(), "Cannot delete a resource with children")) {
```

Comment:
Don't like the hard coded message match, but that's coming from another service so I'm not sure how else to handle it.

Reply:
I totally agree. I don't know another way to do it.